# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/base/nsCaret.h
# Commit: caab023e5d2d
# Full Hash: caab023e5d2d6daee7993353cecc142dde2483fb
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2024-03-11 21:13:39
# Regressor Bug: 1860328
# File Overlap Count: 2
# Description:
#   Bug 1860328 - Fix a bug where caret misses when dragging and dropping r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D192377
# ==============================================================================

diff -r 09b0c2bfb164 -r caab023e5d2d layout/base/nsCaret.h
--- a/layout/base/nsCaret.h	Mon Mar 11 13:36:59 2024 +0000
+++ b/layout/base/nsCaret.h	Mon Mar 11 13:41:27 2024 +0000
@@ -12,6 +12,7 @@
 #include "mozilla/MemoryReporting.h"
 #include "mozilla/dom/Selection.h"
 #include "nsCoord.h"
+#include "nsIFrame.h"
 #include "nsISelectionListener.h"
 #include "nsIWeakReferenceUtils.h"
 #include "nsPoint.h"
@@ -279,6 +280,11 @@
    * it's in non-user-modifiable content.
    */
   bool mIgnoreUserModify;
+
+  /**
+   * mLastCaretFrame is the last caret frame that was scheduled to paint.
+   */
+  WeakFrame mLastCaretFrame;
 };
 
 #endif  // nsCaret_h__